cmake/modules/hunter_install_libsodium_config.cmake

```diff
@@ -0,0 +1,20 @@
+include(CMakePackageConfigHelpers)
+include(hunter_test_string_not_empty)
+
+function(hunter_install_libsodium_config)
+
+    hunter_test_string_not_empty("${HUNTER_libsodium_VERSION}")
+    hunter_test_string_not_empty("${HUNTER_PACKAGE_INSTALL_PREFIX}")
+    hunter_test_string_not_empty("${HUNTER_SELF}")
+
+    set(templates "${HUNTER_SELF}/cmake/templates")
+    set(configs "${HUNTER_PACKAGE_INSTALL_PREFIX}/lib/cmake/libsodium")
+
+    write_basic_package_version_file("${configs}/libsodiumConfig.cmake" VERSION "${HUNTER_libsodium_VERSION}" COMPATIBILITY ExactVersion)
+
+    configure_file(
+            "${templates}/libsodiumConfig.cmake.in"
+            "${configs}/libsodiumConfig.cmake"
```

Comment:
> Unless the release and debug build can be in the same lib file

not sure I understand what does it mean

> the autogenerated cmake isn't handling the case where more than one build config is done at the same time

It does. That's the whole point. It can handle any number of configs.